### DISCUSSION TITLE
MOU restart with chance constraints - tricks?

### DISCUSSION THREAD

**Discussion by cnicol-gwlogic** [usgs/pestpp#304] [Q&A]
Any tricks to get around this error? "MOEAerror: chance constraints not yet supported with restart"
...Or is it as it reads? Wishful thinking maybe. Thanks a lot. 

Bonus question: what's the effective difference if I don't supply the restart obs file, but do provide a decision variable file from a prematurely terminated mou run's latest and greatest generation? Is it simply that I lost that single generation of obs-equivalents that I had previously gotten using that dv_par file? If so, not a big deal at all. But am I missing something conceptually important here?

**Comment by jtwhite79** [comment]
This is a scary question!  Can you give some specifics re when and how the stack is being evaluated?

**Reply by cnicol-gwlogic** [reply]
Ok that doesn't sound great. I'm using single point chance assessment, and recalcing that every 10 generations. Enough info?

**Reply by jtwhite79** [reply]
Which input options are giving mou for this restart?  I thought that restart was only not implemented for multiple chance points...

**Reply by cnicol-gwlogic** [reply]
I tried this (via latest release 5.2.7):
![image](https://github.com/user-attachments/assets/74de812e-7489-44b5-8cd7-5fb26127b0ed)
...dv_pop and obs_pop being the latest generation (28) results across the 100 members.
That gave me the error above. I did wonder why it was a problem, given it immediately recalcs the stack outputs anyway, but I figured there must be some in-memory stuff from previous generations that was important and not saved to disk. Is there anything lingering from previous generations that is used in later ones (that is not saved in those obs_pop / dv_pop files)?

Anyway, don't stress about this query much (thank you though! yet again...) - I think I will re-run that mou batch anyway using chance points 'all' instead. I tried it on my 2nd case (without re-evaluating the stack at all), and I get heaps more feasible members; the pareto front looks sharper in fewer generations too, and seems more "filled out" - and this is for an even more "difficult" / "tight feasibility" mod of my first case. I had thought this might be so, and it proved to be, in my case at least. To make that a little more doable in a shorter timeframe, I culled my stack in half (randomly) - I figure there are always compromises to be made, and this one seems less of a compromise than single chance point assessment, given I am assessing borefield pumping rate dv pars.

Also, I think I was getting a manager PC crash on some later stack re-evaluations (could possibly have been an overheating issue too though) - don't seem to be getting those now when not re-evaluating / using 'all' chance points.

Apologies for the ramble...and thanks for the thoughts, really appreciate it.

**Reply by jtwhite79** [reply]
Hey @cnicol-gwlogic really sorry the delay - scrambling this week.  I was under the hood today there is some support for restarting mou with existing files. Any chance you can share your model+pest files with me so I can take a look?  Seems like for single chance pt we ought to be able to hotstart with existing stack results...

**Reply by cnicol-gwlogic** [reply]
Thanks Jeremy - no need to apologise - totally get it. I need to request permission to share this one, it is politically sensitive and under a confidentiality agreement. I will sus them out and report back here. 

But thanks for the offer, and thanks heaps for looking into the possibility for me.
